lib/Amazon/S3/Thin/SignerV2.pm

```diff
@@ -54,21 +65,30 @@ sub calculate_signature {
     # don't include anything after the first ? in the resource...
     $path =~ /^([^?]*)/;
     $string_to_sign .= "/$1";
+    $path = substr($path, length $1);
```

Comment:
re-assigning the variable $path makes code less readable.
assigning a new variable is better. e.g.  `$query_string` or something. 
(but in general QUERY_STRING does not include  `?` , so we should be careful)